core/src/main/scala/org/apache/spark/sql/execution/row/RowFormatScanRDD.scala

```diff
@@ -411,7 +414,7 @@ final class InternalRowIteratorOnRS(conn: Connection,
       i = i + 1
     }
     moveNext()
-    mutableRow
+    unsafeproj.apply(mutableRow)
```

Comment:
As I mentioned above, ideally this can directly create an UnsafeRow with CodeGeneration, but since it is RowFormatScanRDD so low priority.